src/components/LibraryStats/libraryStatsLogic.ts

```diff
@@ -0,0 +1,61 @@
+import { kea } from 'kea'
+import { librariesList } from './libraries'
+
+export interface Library {
+    name: string
+    path: string
+    lastCommit: string
+    stars: number
+    openIssues: number
+    pullRequests: number
+}
+
+export const libraryStatsLogic = kea({
+    loaders: {
+        libraries: [
+            [] as Library[],
+            {
+                loadLibraryStats: async () => {
+                    for (let i = 0; i < librariesList.length; ++i) {
+                        const library = librariesList[i]
+                        const repoRes = await (await fetch(`https://api.github.com/repos/${library.path}`)).json()
+
+                        librariesList[i]['stars'] = repoRes.stargazers_count
+                        librariesList[i]['openIssues'] = 0
+                        librariesList[i]['pullRequests'] = 0
+                        librariesList[i]['lastCommit'] = '-'
+
+                        if (typeof repoRes.open_issues_count === 'number' && repoRes.open_issues_count > 0) {
+                            const pullRequestsRes = await (
+                                await fetch(`https://api.github.com/repos/${library.path}/pulls?state=open`)
+                            ).json()
```

Comment:
Github has a 60 requests per hour limit on their free API. This will go under that, but visiting the page a few times will bust this limit. Not sure if there's a good way to cache this data to avoid the limit. Could we get some or most of the info already in the `/repos` endpoint? PR count probably not, but the last commit could be taken from `pushed_at` perhaps?

Reply:
Yeah I'm aware of this, didn't spend too much time on it given effort tradeoffs (i.e. didn't think it was extremely important). There are a few approaches like:

- `localStorage` with a timestamp
- Fetch all this stuff at build time (assuming daily builds this will be reasonably up-to-date, but not immediate)

Reply:
Indeed, `pushed_at` we can use to substitute the commits fetch I believe

Reply:
Added a cache via `localStorage` - quickest to implement and should work in most cases.

Reply:
Nice